daemons/dss-sync-sfn/app.py

```diff
@@ -0,0 +1,306 @@
+                "Next": "Quit"
+            }],
+            "Default": "CheckDeps"
+        },
+        "WaitForDeps": {
+            "Type": "Wait",
+            "SecondsPath": "$.sleep_seconds",
+            "Next": "CheckDeps"
+        },
+        "CheckDeps": {
+            "Type": "Task",
+            "Resource": None,  # This will be set by Domovoi to the Lambda ARN
+            "Next": "WaitOrCopyOrQuit",
+            "Retry": retry_config
+        },
+        "OneshotCopy": {
+            "Type": "Task",
+            "Resource": None,  # This will be set by Domovoi to the Lambda ARN
+            "End": True,
+            "Retry": retry_config
+        },
+        "MultipartCopyThreadpool": {
+            "Type": "Parallel",
+            "Branches": [],  # This will be filled in with an array of "thread" state machines below
+            "Next": "ComposeParts"
+        },
+        "ComposeParts": {
+            "Type": "Task",
+            "Resource": None,  # This will be set by Domovoi to the Lambda ARN
+            "End": True,
+            "Retry": retry_config
+        },
+        "Quit": {
+            "Type": "Pass",
+            "End": True
+        }
+    }
+}
+
+sfn_thread = {
+    "StartAt": "Worker{t}",
+    "States": {
+        "Worker{t}": {
+            "Type": "Task",
+            "Resource": None,  # This will be set by Domovoi to the Lambda ARN
+            "Next": "Branch{t}",
+            "Retry": retry_config
+        },
+        "Branch{t}": {
+            "Type": "Choice",
+            "Choices": [{
+                "Variable": "$.finished",
+                "BooleanEquals": True,
+                "Next": "EndThread{t}"
+            }],
+            "Default": "Worker{t}"
+        },
+        "EndThread{t}": {
+            "Type": "Pass",
+            "End": True
+        }
+    }
+}
+
+@app.step_function_task(state_name="DispatchSync", state_machine_definition=sync_sfn_num_threads)
+def dispatch_sync(event, context):
+    """
+    Processes the storage event notification and orchestrates the rest of the copying:
+    - If the notification is for a file blob:
+        - If the blob is under the one-shot threshold, immediately copies it and exits the state machine.
+        - Otherwise, configures the state machine to run the threadpool (copy blob parts) and closer (compose the copy).
+    - Otherwise, check if all referenced entities are already copied to the destination:
+        - For a file manifest, check that the blob is there.
+        - For a bundle manifest, check that file manifests for all files in the bundle are there.
+        - For a collection manifest, check that all collection contents are there.
+        If the checks fail, cause the state machine to sleep for 8 seconds, then try again.
+        If the checks succeed, do a one-shot copy of the manifest.
+    """
+    if event["source_key"].startswith("blobs"):
+        if int(event["source_obj_metadata"]["size"]) > AWS_MIN_CHUNK_SIZE:
```

Comment:
The AWS oneshot copy threshold is lower. It would be a logic fault to not trigger the multipart copy code above the AWS oneshot copy threshold, but below the GCP oneshot copy threshold.